dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Views/Patient/Index.cshtml

```diff
@@ -54,25 +51,43 @@
                     <tr>
                         <td class="ms-thumbnail col-md-1"><img class="img-thumbnail" src="@task.TaskImageUrl" alt="Task thumbnail icon" /></td>
                         <td class="col-md-9 @(task.ScheduleType == TimelineScheduleType.Unscheduled ? "occurrence" : "")">
-                            @task.TaskName<br />
-                            @TaskTime(task)
+                            @if (task.ScheduleType == TimelineScheduleType.Unscheduled)
+                            {
+                                @task.TaskName
+                            }
+                            else
+                            {
+                                @Html.ActionLink(task.TaskName, "TaskOccurrence", new {id = task.TaskId, startDate = Model.StartDate, endDate = Model.EndDate, personId = Request.Params["personId"], recordId = Request.Params["recordId"]})
+                            }
+                            <br/>@TaskTime(task)
+                        </td>
+                        <td class="col-md-2">
+                            @if (task.ScheduleType != TimelineScheduleType.Unscheduled)
+                            {
+                                if (task.CompletionMetrics.CompletionType == ActionPlanTaskCompletionType.Scheduled && task.OccurrenceCount > 0)
+                                {
+                                    @:Done!
+                                }
+                                else if (task.CompletionMetrics.CompletionType == ActionPlanTaskCompletionType.Frequency)
+                                {
+                                    @Html.Raw($"{task.OccurrenceCount}/{task.CompletionMetrics.RequiredNumberOfOccurrences.Value}<br/>{task.CompletionMetrics.RecurrenceType}")
```

Comment:
In razor, isn't there an easier way to output the result of a calculation? Would `@(...)` not work?